tools/ci_build/github/azure-pipelines/templates/get-docker-image-steps.yml

```diff
@@ -74,10 +71,11 @@ steps:
   - task: Cache@2
     displayName: Cache Docker Image Task
     inputs:
-      key: ' "${{ parameters.Repository }}" | ${{ parameters.DockerCacheKeys }} '
+      key: ' "${{ parameters.Repository }}" | "$(Build.SourceVersion)" '
```

Comment:
I don't understand. Does it mean whenever "$(Build.SourceVersion)" is changed, the cache is invalid? Then almost all the builds have different "$(Build.SourceVersion)".

Reply:
And I'm not very sure what does $(Build.SourceVersion) mean.  It seems not reliable.

Reply:
1. $(Build.SourceVersion) used to create new cache. source is changed, the cache should be updated.
2.   restoreKeys is used to get the cache from the server
```
      restoreKeys: |
        "${{ parameters.Repository }}" | "$(Build.SourceVersion)"
        "${{ parameters.Repository }}"
```
      if source isn't changed, it will get the cache of `"${{ parameters.Repository }}" | "$(Build.SourceVersion)"`
     if source is changed, there isn't cache of `"${{ parameters.Repository }}" | "$(Build.SourceVersion)"` in server, it will get the latest cache under "${{ parameters.Repository }}"

Reply:
From the log https://dev.azure.com/onnxruntime/2a773b67-e88b-4c7f-9fc0-87d31fea8ef2/_apis/build/builds/916595/logs/14

Resolving key:
 - "onnxruntimecpubuildaten"                [string]
 - **8e3f5f20873e57a689bbc65872e7aed2ceb52818** [string]
Resolved to: "onnxruntimecpubuildaten"|8e3f5f20873e57a689bbc65872e7aed2ceb52818

Fingerprint: `"onnxruntimecpubuildaten"|8e3f5f20873e57a689bbc65872e7aed2ceb52818`
Fingerprint: `"onnxruntimecpubuildaten"|8e3f5f20873e57a689bbc65872e7aed2ceb52818|**`
Fingerprint: `"onnxruntimecpubuildaten"|**`
There is a cache hit: `"onnxruntimecpubuildaten"|**b7fce71e033a23c52cf2117cb502631b9a1f5695**`
Used scope: 194;2a773b67-e88b-4c7f-9fc0-87d31fea8ef2;refs/heads/zhanyi/cacheimage1;Microsoft/onnxruntime
Entry found at fingerprint: `"onnxruntimecpubuildaten"|b7fce71e033a23c52cf2117cb502631b9a1f5695`

The cache hit is **b7fce71e033a23c52cf2117cb502631b9a1f5695**, not **8e3f5f20873e57a689bbc65872e7aed2ceb52818**

Reply:
$(Build.SourceVersion) is widely used.
https://github.com/microsoft/azure-pipelines-tasks/issues/13666#issuecomment-704423419
and ADO supports it.
https://developercommunity.visualstudio.com/t/azure-pipeline-has-incorrect-commit-id-for-buildso/1195934

Reply:
> Commit id in "Build.SourceVersion" is before merge with main branch or after merge?
It‘s before merge. It's the commit id in fact.